autokeras/auto/auto_model.py

```diff
@@ -214,9 +153,107 @@ def _add_node(self, input_node):
         if input_node not in self._node_to_id:
             self._node_to_id[input_node] = len(self._node_to_id)
 
+    def fit(self,
+            x=None,
+            y=None,
+            validation_data=None,
+            **kwargs):
+        # Initialize HyperGraph model
+        x = layer_utils.format_inputs(x, 'train_x')
+        y = layer_utils.format_inputs(y, 'train_y')
+
+        # TODO: Set the shapes only if they are not provided by the user when
+        #  initiating the HyperHead or Block.
+        for x_input, input_node in zip(x, self.inputs):
+            input_node.shape = x_input.shape[1:]
+        for y_input, output_node in zip(y, self.outputs):
+            if len(y_input.shape) == 1:
+                y_input = np.reshape(y_input, y_input.shape + (1,))
+            output_node.shape = y_input.shape[1:]
+            output_node.in_hypermodels[0].output_shape = output_node.shape
+
+        # Prepare the dataset
+        if validation_data is None:
```

Comment:
`validation_split` and `validation_data` are already valid arguments in `model.fit`. Besides, handling validation data correctly is very complicated (here you assume numpy arrays, but this is only one of the possible cases). It is much simpler to just pass the fit_args and fit_kwargs blindly to `model.fit()` without any special handling of the validation data.

Reply:
If we don't handle it, does it mean the users have to provide the validation data? If so we would need to handle it even it is complicated.